src/zoid/buttons/util.js

```diff
@@ -357,3 +360,60 @@ export function getButtonSize(
     }
   }
 }
+
+export const getModal: (
+  clientID: string,
+  merchantID: $ReadOnlyArray<string> | void
+) => Object = memoize(async (clientID, merchantID) => {
+  try {
+    const namespace = getNamespace();
+    if (!window[namespace].MessagesModal) {
+      const modalBundleUrl = () => {
+        let envPiece;
+        switch (getEnv()) {
+          case "test":
+            return "/base/test/integration/windows/button/modal.js";
+          case "local":
+          case "stage":
+            envPiece = "stage";
+            break;
+          case "sandbox":
+            envPiece = "sandbox";
+            break;
+          case "production":
+          default:
+            envPiece = "js";
+        }
+        return `https://www.paypalobjects.com/upstream/bizcomponents/${envPiece}/modal.js`;
+      };
```

Comment:
Ah yeah that's a good callout for the test case. All of the different SDK components have different `globals.js` that get merged together, so introducing another top level namespace could potentially cause a collision. We may want to stay inside `__PAYPAL_CHECKOUT__` since that's what's already reserved here. If we want to include the full URL though that may likely be fine. We can make that change and then see how the rest of the team feels.